sdk/typescript/src/cryptography/keypair.ts

```diff
@@ -7,12 +7,26 @@ import { blake2b } from '@noble/hashes/blake2b';
 import { bcs } from '../bcs/index.js';
 import { IntentScope, messageWithIntent } from './intent.js';
 import type { PublicKey } from './publickey.js';
-import type { SignatureScheme } from './signature-scheme.js';
+import { SIGNATURE_SCHEME_TO_FLAG, type SignatureScheme } from './signature-scheme.js';
 import type { SerializedSignature } from './signature.js';
 import { toSerializedSignature } from './signature.js';
+import { bech32 } from 'bech32';
 
 export const PRIVATE_KEY_SIZE = 32;
 export const LEGACY_PRIVATE_KEY_SIZE = 64;
+export const SUI_PRIVATE_KEY_PREFIX = "suiprivkey";
+
+export function encodeSuiKeyPair(bytes: Uint8Array, scheme: SignatureScheme): string {
```

Comment:
This method doesn't really fit well with how the rest of this file is doing.

I think what I'd do instead is update the `Keypair` class to take the signature scheme, then remove the export from the concrete keypair classes, and implement it in the abstract class instead, inlining the logic from this function into that export method instead.

You'd probably need to add a new abstract method for getting the private key bytes instead, although that could also just be passed to the constructor instead.

cc @Jordan-Mysten

Reply:
good point, revised to implement an abstract export(), and added getSecretKeyBytes() to Keypair interface